tmt/base.py

```diff
@@ -2390,8 +2398,6 @@ def import_plan(self) -> Optional['Plan']:
         self._imported_plan = Plan(node=node, run=self.my_run, logger=self._logger)
         self._imported_plan._original_plan = self
 
-        # Update imported plan environment with the local environment
-        self._imported_plan.environment.update(self.environment)
```

Comment:
This line was nasty. `self._imported_plan.environment` is actually the `environment` method with `@property` decorator. It used to work just fine when the method always returned the same `self._environment` object, the `update()` call would then update the object just fine, making the changes persistent. When I removed `self._environment` and changed `environment` method to always return a fresh object, this line stopped working. Is this a bug or something that was intended?

I changed this functionality to pass this data the other way. In `environment` method, the imported plan fetches the environment from `self._original_plan.environment`.

Reply:
Yeah, agreed, this seems like a better approach. Thanks for improving this. Looking closer at the `self.my_run.environment` I see there is also environment extracted from the workdir so I guess that line cannot be dropped.